web_server/posts/views.py

```diff
@@ -1,135 +1,86 @@
-    if request.method not in methods:
-        response = HttpResponse(f"Attempt to request endpoint using '{request.method}' method, "
-                                f"but only one of methods in '{list(methods.keys())}' allowed.",
-                                status=405)
-        response["Allow"] = ", ".join(methods.keys())
-        return response
-
-    # Check if this method can produce the requested content type
-    accepted_content_types = request.headers['Accept'].split(',')
-    accepted_type = None
-    for content_type in accepted_content_types:
-        # Find and serve the first acceptable type to the user
-        # @todo do we need to do actual pattern matching for things like 'application/*'
-        if content_type == '*/*' and len(methods) > 0:
-            # The user will accept anything, serve the first available type on this method
-            accepted_type = list(methods[request.method].keys())[0]
-            break
-        if content_type in methods[request.method]:
-            accepted_type = content_type
-            break
-
-    if accepted_type is None:
-        return HttpResponse(f"Server unable to produce content of type {accepted_type}. "
-                            f"Available content types one of {list(methods[request.method].keys())}.",
-                            status=406)
-
-    # Attempt to fulfill the request using the handler
-    try:
-        response = methods[request.method][accepted_type](request)
-        if issubclass(type(response), HttpResponse):
-            return response
-        else:
-            raise Exception("Response handler unable to produce HttpResponse like object")
-    except Exception as e:
-        return HttpResponse(f"The server failed to handle your request. Cause Hint: {e}", status=500)
-
-# retrieve all posts marked as public on the server
 def retrieve_all_public_posts_on_local_server(request):
     """
     For endpoint http://service/posts
     Retrieves all public posts located on the server
     Methods: GET
-    :param request: should specify Accepted content-type, default is HTML
+    :param request: should specify Accepted content-type
     :returns: application/json | text/html
     """
-    default_page_size = "10"
-    max_page_size = 50
-    def json_handler(req):
-        # @todo: how to extract this pagination behavior so that other endpoints can use it automatically?
-        page = int(req.GET.get('page', "1"))
-        size = min(int(req.GET.get('size', default_page_size)), max_page_size)
-        posts = Post.objects.filter(visibility="PUBLIC")
-        pager = Paginator(posts, size)
+    def json_handler(posts, pager, pagination_uris):
+        # Use Django's serializers to encode the posts as a python object
+        json_posts = loads(serializers.serialize('json', posts))
+
+        # Explicitly add authors to the serialization
+        # @todo should we reduce the number of fields that get served here?
```

Comment:
I am looking into how to reduce the fields

Reply:
I know that you can pass in a fields arg, but then we would have to list out all the fields we want when in reality it would be nicer to just list out the two or three (password field I'm looking at you) fields we _didn't_ want.

Reply:
May as well just do it the long way

Reply:
Very well. I've implemented a custom serializer derived from the JSON serializer that implements blacklisting instead of whitelisting. 😎👉👉